lib/core/test/dict_test.cpp

```diff
@@ -0,0 +1,538 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+#include <gtest/gtest.h>
+
+#include "scipp/core/dict.h"
+#include "scipp/units/dim.h"
+
+#include "test_macros.h"
+
+using namespace scipp::core;
+using scipp::Dim;
+
+using DimDict = Dict<Dim, int>;
```

Comment:
There are a bunch of places that use `std::move` on the values of the dict, is it worth testing with a `Value` that has a non-trivial `move`, such as `std::string`?

Reply:
Good point. But `std::string` is not the best choice because its move might be trivial. I'm going to use a custom type that holds an int behind a pointer.